napalm-ruckus-fastiron/fi.py

```diff
@@ -0,0 +1,1378 @@
+        temp_list = list()                          # Creates a list to store variables
+        for a_string in nline_list:                 # iterates through list
+            if a_string .__contains__(del_word):    # if word matches, word is skipped
+                continue
+            else:
+                temp_list.append(a_string.split())  # Word didn't match store in list
+        return temp_list
+
+    @staticmethod
+    def facts_uptime(my_string):  # TODO check for hours its missing....
+        my_list = ["day(s)", "hour(s)", "minute(s)", "second(s)"]   # list of words to find
+        my_pos = [-1, -1, -1, -1]                   # relative position of interest
+        total_seconds = 0                           # data variables
+        multiplier = 0
+        t_dictionary = FastIronDriver.find_words(my_string, my_list, my_pos)    # retrieves pos
+
+        for m in t_dictionary.keys():               # Checks word found and its multiplier
+            if m == "second(s)":                    # converts to seconds
+                multiplier = 1
+            elif m == "minute(s)":
+                multiplier = 60
+            elif m == "hour(s)":
+                multiplier = 3600
+            elif m == "day(s)":
+                multiplier = 86400
+            total_seconds = int(t_dictionary.get(m))*multiplier + total_seconds
+        return total_seconds
+
+    @staticmethod
+    def facts_model(string):
+        model = FastIronDriver.retrieve_all_locations(string, "Stackable", 0)[0]
+        return model                                # returns the model of the switch
+
+    @staticmethod
+    def facts_hostname(string):
+        if string.__contains__("hostname"):
+            hostname = FastIronDriver.retrieve_all_locations(string, "hostname", 0)[0]
+            return hostname                         # returns the hostname if configured
+        else:
+            return None
+
+    @staticmethod
+    def facts_os_version(string):
+        os_version = FastIronDriver.retrieve_all_locations(string, "SW:", 1)[0]
+        return os_version                           # returns the os_version of switch
+
+    @staticmethod
+    def facts_serial(string):
+        serial = FastIronDriver.retrieve_all_locations(string, "Serial", 0)[0]
+        serial = serial.replace('#:', '')
+        return serial                               # returns serial number
+
+    @staticmethod
+    def facts_interface_list(shw_int_brief, pos=0, del_word="Port", trigger=0):
+        interfaces_list = list()
+        n_line_output = FastIronDriver.creates_list_of_nlines(shw_int_brief)
+        interface_details = FastIronDriver.delete_if_contains(n_line_output, del_word)
+
+        for port_det in interface_details:
+
+            if trigger == 0:
+                interfaces_list.append(port_det[pos])
+            else:
+                if port_det[pos].__contains__("ve") or port_det[pos].__contains__("lb") or \

```

Comment:
```python
if any(x in port_det[pos] for x in ["ve", "lb", "tunnel"):
     continue
```

Reply:
Thank you to introducing me to any, this has been changed :)